Test.cs

```
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+
+namespace ConsoleApp1
+{
+	using System.IO;
+
+	/// <summary>
+	/// Class to read in a csv and parse the values into type
+	/// </summary>
+	public class TestClass
+	{
+		public string ErrorMessage;
+
+		private object lockA = new object();
+
+		private object lockB = new object();
+
+		private IResultValidator resultValidator = new ResultValidator();
+
+		public List<Result> GetResult()
+		{
+			// Current path to file.
+			var fp = "C:\\Temp\\FileToOpen.csv";
+
+			var text = File.ReadAllText(fp);
+
+			List<Result> results = new List<Result>(100);
+
+			lock (lockA)
+			{
+				foreach (var l in text.Split(char.Parse(",")))
+				{
+					try
+					{
+						if (l.StartsWith("ExcepectedValue"))
+						{
+							lock (lockB)
+							{
+								var newResult = this.GetResult(l);
+								if (this.resultValidator.IsValid(newResult))
+								{
+									results.Add(newResult);
+								}
+							}
+						}
+					}
+					catch (Exception)
+					{
+					}
```

No point in having "try" with empty catch blocks. Either handle or let it bubble up.